localstack/aws/connect.py

```diff
@@ -333,7 +333,7 @@ def _get_client(
                 aws_access_key_id=aws_access_key_id,
                 aws_secret_access_key=aws_secret_access_key,
                 aws_session_token=aws_session_token,
-                config=config,
+                config=config.merge(Config(retries={"max_attempts": 0})),
```

Comment:
Would this be reverted? Are we trying to address an issue with the test suite? I would rather set this in the fixtures.

Reply:
No this is also intended to be set for internal communication, though we should observe this a bit. Generally I think it makes more sense to fail fast if there's an issue instead of retrying countless times.

Reply:
We should not hide potential issues through retries for internal calls as long as we assume that LocalStack runs on the same machine (i.e., without a potentially problematic network connection).

Reply:
To keep this a bit less controversion for now, I've switched the default back to normal retries & explicitly set `DISABLE_BOTO_RETRIES` in our CI tests. This means you'll need to set this in your environment locally though if you want to execute your tests without retries.